fl4health/clients/tabular_data_client.py

```diff
@@ -0,0 +1,115 @@
+from fl4health.utils.metrics import Metric
+
+
+class TabularDataClient(BasicClient):
+    def __init__(
+        self, data_path: Path, metrics: Sequence[Metric], device: torch.device, id_column: str, target_column: str
+    ) -> None:
+        super().__init__(data_path, metrics, device)
+        self.parameter_exchanger = FullParameterExchanger()
+        self.tabular_features_info_encoder: TabularFeaturesInfoEncoder
+        self.tabular_features_preprocessor: TabularFeaturesPreprocessor
+        self.df: pd.DataFrame
+        self.input_dimension: int
+        self.output_dimension: int
+        self.id_column = id_column
+        self.target_column = target_column
+        # The aligned data and targets, which are used to construct dataloaders.
+        self.aligned_features: NDArray
+        self.aligned_targets: NDArray
+
+    def setup_client(self, config: Config) -> None:
+        """
+        Initialize the client by encoding the information of its tabular data
+        and initializing the corresponding TabularFeaturesPreprocessor.
+
+        config[FORMAT_SPECIFIED] indicates whether the server has obtained
+        the source of information to perform feature alignment.
+        If it is True, it means the server has obtained such information
+        (either a priori or by polling a client).
+        So the client will encode that information and use it instead
+        to perform feature preprocessing.
+        """
+        assert FORMAT_SPECIFIED in config.keys()
+        format_specified = self.narrow_config_type(config, FORMAT_SPECIFIED, bool)
+
+        # Encode the information of the client's local tabular data.
+        self.df = self.get_data_frame(config)
+        self.tabular_features_info_encoder = TabularFeaturesInfoEncoder.encoder_from_dataframe(
+            self.df, self.id_column, self.target_column
+        )
+
+        if format_specified:
+            # Since the server has obtained its source of information,
+            # the client will encode that instead.
+            self.tabular_features_info_encoder = TabularFeaturesInfoEncoder.from_json(
+                self.narrow_config_type(config, FEATURE_INFO, str)
+            )
+            self.tabular_features_preprocessor = TabularFeaturesPreprocessor(self.tabular_features_info_encoder)
+
+            # preprocess features.
+            self.aligned_features, self.aligned_targets = self.tabular_features_preprocessor.preprocess_features(
+                self.df
+            )
+
+            # Obtain the input and output dimensions to be sent to
+            # the server for global model initialization.
+            self.input_dimension = self.aligned_features.shape[1]
+            target_type = self.tabular_features_info_encoder.get_target_type()
+            if target_type == ORDINAL or target_type == BINARY:
+                self.output_dimension = len(self.tabular_features_info_encoder.get_target_categories())
```

Comment:
Rather than performing the `len(...)` here, maybe we can bury this as another function of `tabular_features_info_encoder` as `get_target_dimension()`

Reply:
Yes, good call